microprofile/cors/src/main/java/io/helidon/microprofile/cors/CorsCdiExtension.java

```diff
@@ -87,6 +115,14 @@ void ready(@Observes @Priority(LIBRARY_BEFORE + 10) @Initialized(ApplicationScop
                             CrossOrigin.class.getSimpleName(),
                             methodsWithCrossOriginIncorrectlyUsed));
         }
+        if (clazzWithClassLevelAnnotationCrossOriginWithoutOptionMethods != null) {
+            throw new IllegalArgumentException(
+                    String.format(
+                            "%s has class-level annotation @%s and should have at least one @%s method",
+                            clazzWithClassLevelAnnotationCrossOriginWithoutOptionMethods.getSimpleName(),
+                            CrossOrigin.class.getSimpleName(),
+                            OPTIONS.class.getSimpleName()));
+        }
```

Comment:
See earlier comment. I think this checking, if it happens, needs to be inside (or invoked from) `processManagedBean`.